config/config.go

```diff
@@ -224,6 +225,10 @@ type Config struct {
 	StorageConfig     StorageConfig   `yaml:"storage,omitempty"`
 	TracingConfig     TracingConfig   `yaml:"tracing,omitempty"`
 
+	// Secret management:
+	kubernetes.ClientConfig  `yaml:"kubernetes_sp_config,omitempty"`
```

Comment:
non-blocker for fork: It's a bit of code smell to put provider specific thingy on the top of config.

I guess it's that decision to either have deduplicate client configuration and separate secret references etc.

But honestly even with your main proposal I would expect ["secretProviders"](https://docs.google.com/document/d/1EqHd2EwQxf9SYD8-gl3sgkwaU6A10GhiN7aw-2kx7NU/edit#heading=h.qndjfteihpa) field first.

I also noticed https://docs.google.com/document/d/1EqHd2EwQxf9SYD8-gl3sgkwaU6A10GhiN7aw-2kx7NU/edit#heading=h.shix1mn6vcdl idea would probably be simpler to implement.

Still up to you, we can keep as it is for now, just want to leave some feedback for upstream work.

Reply:
Intentional for fork! Thanks!